### ISSUE TITLE
How to use pdb to see what happened step

### ISSUE THREAD

**Issue by zhijiesun** [facebookresearch/PyTorch-BigGraph#186]
## Steps to reproduce

1. When I am training, I want to see what happened step by step like using torchbiggraph_train command.

## Expected Results

Can I use library like pdb to see the produce of training?

**Comment by lw** [comment]
PDB is not a supported way to inspect training, and it will be very hard to use due to the fact that PBG makes use of multiple processes.

If the information logged by the training script isn't enough you could modify it to suit your needs. Note also that you can do some offline real-time tracking by picking up the checkpoints that are saved and evaluating those.

**Comment by zhijiesun** [comment]
OK. Thank you for your kindly reply.

**Comment by zhijiesun** [comment]
Can I config 1 worker when training to understand the process of training. still use PDB

**Comment by lw** [comment]
I don't think using one worker only will help, as it would still spawn a subprocess, but feel free to try.

If you're just curious of how PBG works I recommend reading the doc and the code itself. If you are having unexpected results and want to root cause them maybe it's better to explain what problem you're facing.

**Comment by zhijiesun** [comment]
Hi Luca, I config ranking loss with margin equal to 0.1. The training curve is perfect but evaluation curve goes up all at once at around 4 epoch, I don't know why. I wonder if you have ever encountered such a problem before? or some suggestion on how to debug it? Thank you.
